### PR TITLE
Allow cloud instances for branches

### PR DESCRIPTION
**Pull request by quinchs**
Cloud instances don't have a `credentials.json` file to store the current branch information in, instead we allow cloud instances to be used *if* they're mapped to a project, we then update the `database` file in that projects stash directory to keep track of the current branch.

### CODE REVIEW

File: src/branch/context.rs [hunk 1]
```diff
@@ -19,44 +22,55 @@ pub struct Context {
...
+        // read from credentials
+        if project_dir.is_some()
+            && instance_name.as_ref().map_or(false, |v| match v {
+                InstanceName::Local(_) => true,
+                _ => false,
+            })
+        {
+            let instance_name = match instance_name.as_ref().unwrap() {
+                InstanceName::Local(instance) => instance,
+                unsupported => anyhow::bail!(format!("Cannot use instance type {}", unsupported)),
```
Comment (elprans) [inline src/branch/context.rs:56]: A better error message is needed:

```suggestion
                unsupported => anyhow::bail!(format!("cannot switch branches on Cloud instance {org}/{slug}: instance is not linked to a project", unsupported)),
```

Comment (elprans) [reply src/branch/context.rs:56]: Wait, why does this apply to _all_ branch commands?  Shouldn't only the `switch` be affected?

**Comment by quinchs** [comment]
> Wait, why does this apply to all branch commands? Shouldn't only the switch be affected?

I guess it can work for non switch/rename commands, assuming the database default branch

**Comment by quinchs** [comment]
Alright, all commands work for cloud instances except switch/rename if theres no project linked to the instance

**Comment by quinchs** [comment]
if you rename the current branch, it will preform a switch to the new name

**Comment by quinchs** [comment]
> So only gate the switching part

thats what this PR does now

**Comment by elprans** [comment]
Please fix `rustfmt`

**Review by elprans — APPROVED** [review]

**Pull request merged**